### PR TITLE
Allow MongoDB connections to use TLS

### LINKED ISSUE #205: Unable to connect to Azure's Cosmos DB (MongoDB)
Azure runs Cosmos DB on a non-standard MongoDB port, and requires SSL. Due to these two things docker_auth is unable to connect to the MongoDB server.

Example of the config block that I attempted to use to connect so far:
```
mongo_auth:
  dial_info:
    addrs: ["docker-registry-auth-test.documents.azure.com:10255"]
    timeout: "10s"
    database: "docker_auth"
    username: "docker-registry-auth-test"
    password: "abcd1234"
  collection: "users"
```
Azure provides a connection string similar to: `mongodb://docker-registry-auth-test:abcd1234@docker-registry-auth-test.documents.azure.com:10255/?ssl=true`

### PR DESCRIPTION
**Pull request by techknowlogick**
Fixes #205

---
This change is [Reviewable](https://reviewable.io/reviews/cesanta/docker_auth/207)

### CODE REVIEW

File: auth_server/mgo_session/mgo_session.go [hunk 1]
```diff
@@ -30,6 +31,7 @@ import (
 type Config struct {
 	DialInfo     mgo.DialInfo `yaml:",inline"`
 	PasswordFile string       `yaml:"password_file,omitempty"`
+	EnableTls    bool         `yaml:"enable_tls,omitempty"`
```
Comment (rojer) [inline auth_server/mgo_session/mgo_session.go]: `EnableTLS`

File: auth_server/mgo_session/mgo_session.go [hunk 2]
```diff
@@ -61,6 +63,12 @@ func New(c *Config) (*mgo.Session, error) {
 		c.DialInfo.Password = strings.TrimSpace(string(passBuf))
 	}
 
+	if c.EnableTls == true {
```
Comment (rojer) [inline auth_server/mgo_session/mgo_session.go]: no need to compare to `true` explicitly, just `if c.EnableTLS {`

**Comment by rojer** [comment]
thank you!

**Comment by techknowlogick** [comment]
@rojer Thanks for this project (and the merge). It is very helpful.

**Pull request merged**